diff --git a/sepolicy/base/public/domain.te b/sepolicy/base/public/domain.te
index ed5fed7ca..be0d2f3aa 100644
--- a/sepolicy/base/public/domain.te
+++ b/sepolicy/base/public/domain.te
@@ -221,9 +221,9 @@ neverallow { domain -init -ueventd -riladapter_host debug_only(`-softbus_server'
 #neverallow { domain -ueventd -riladapter_host } dev_file:sock_file *;
 
 neverallow { domain -kernel -init -chipset_init -misc -updater_sa -storage_daemon -partitionslot_host updater_only(`-updater ') -updater_binary -dev_attr_violator -sys_installer_sa -write_updater -rgm_violator_ohos_dev_blk_file -module_update_service
-        -render_service -codec_host -bootanimation -camera_host -location_host -riladapter_host } dev_attr:blk_file { open read write };
+        -render_service -codec_host -bootanimation -camera_host -location_host -riladapter_host -system_basic_hap } dev_attr:blk_file { open read write };
 neverallow { updater_sa sys_installer_sa write_updater } {dev_attr -updater_block_file -dev_block_file}:blk_file { open read write };
-neverallow { render_service codec_host bootanimation camera_host location_host riladapter_host } {dev_attr -sml_block_file }:blk_file { open read write };
+neverallow { render_service codec_host bootanimation camera_host location_host riladapter_host system_basic_hap } {dev_attr -sml_block_file }:blk_file { open read write };
 neverallow { module_update_service } {dev_attr -dev_block_file}:blk_file { open read write };
 # fs operation limit
 neverallow { domain -filesystem_violator -clouddiskservice } *:filesystem ~{ getattr mount remount unmount relabelfrom relabelto quotaget quotamod };
diff --git a/sepolicy/base/public/hap_domain.te b/sepolicy/base/public/hap_domain.te
index a1b9c7be1..f124b91f8 100644
--- a/sepolicy/base/public/hap_domain.te
+++ b/sepolicy/base/public/hap_domain.te
@@ -37,7 +37,7 @@ neverallow system_basic_hap_attr { domain -system_basic_hap_attr }:file never_rw
 neverallow { hap_domain -system_basic_hap_attr -hap_domain_kernel_violators } system_basic_hap_attr:file never_rw_file;
 neverallow { domain -system_basic_hap_attr -processdump debug_only(`-hiperf -native_daemon') } system_basic_hap_attr:process ptrace;
 
-neverallow hap_domain ~{ tmpfs_data_file dev_file proc_attr tmpfs system_bin_file toybox_exec data_user_file hmdfs hap_domain rootfs lib_file system_lib_file vendor_lib_file sysfs_attr etc_file hap_domain_lnk_file_operation_viloator developer_only(`hnp_file_attr hnp_file private_hnp_file') hnp_file_attr_violator}:lnk_file *;
+neverallow { hap_domain -system_basic_hap } ~{ tmpfs_data_file dev_file proc_attr tmpfs system_bin_file toybox_exec data_user_file hmdfs hap_domain rootfs lib_file system_lib_file vendor_lib_file sysfs_attr etc_file hap_domain_lnk_file_operation_viloator developer_only(`hnp_file_attr hnp_file private_hnp_file') hnp_file_attr_violator}:lnk_file *;
 neverallow hap_domain { proc_attr tmpfs system_bin_file rootfs lib_file system_lib_file vendor_lib_file sysfs_attr etc_file developer_only(`hnp_file_attr') hnp_file_attr_violator}:lnk_file ~{ getattr read };
 neverallow hap_domain hap_domain:lnk_file ~{ read_file_perms };
 neverallow { hap_domain -hap_domain_lnk_file_violators } data_user_file:lnk_file *;
@@ -102,8 +102,8 @@ neverallow hap_domain vendor_file_attr:dir never_write_dir;
 neverallow hap_domain { vendor_file_attr -vendor_lib_file -vendor_etc_vulkan_file -vendor_etc_graphic_xengine_file_violator_dir_open_read_search -violator_hap_domain_file_sock_file }:{ file fifo_file lnk_file sock_file } *;
 
 #hap never access blk_file.
-neverallow hap_domain dev_attr:blk_file open;
-neverallow hap_domain dev_attr:blk_file read;
+neverallow { hap_domain -system_basic_hap } dev_attr:blk_file open;
+neverallow { hap_domain -system_basic_hap } dev_attr:blk_file read;
 neverallow hap_domain dev_attr:blk_file write;
 neverallow hap_domain dev_attr:blk_file ioctl;
 
